### PR TITLE
[DDCE-1797]:Charities: Implement new 'amend address' pages

### PR DESCRIPTION
**Pull request by anburoach**
# DDCE-(StoryNumberHere)

Please include a summary / description of the change and which issue it fixes.

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Review by yangwong9664 — CHANGES_REQUESTED** [review]
4 boolean simplify expression warnings can be fixed when compiling project

File: app/models/addressLookup/AddressModel.scala [hunk 1]
```diff
@@ -35,3 +63,29 @@ object AddressModel {
...
+case class AmendAddressModel(line1: String,
+                             line2: Option[String],
+                             line3: Option[String],
+                             town: String,
+                             postcode: String,
+                             country: String) {
+
+  def toConfirmableAddress(implicit messages: Messages): AddressModel =
+    AddressModel(
+      Seq(line1) ++ line2.map(line2 => line2).toList ++ line3.map(line3 => line3).toSeq ++ Seq(town),
```
Comment (yangwong9664) [inline app/models/addressLookup/AddressModel.scala:81]: the .maps are not needed, also is .toList needed for any special reason or can it be .toSeq, for consistency?

**Review by yangwong9664 — CHANGES_REQUESTED** [review]

File: conf/messages.cy [hunk 2]
```diff
@@ -1173,6 +1209,11 @@ authorisedOfficialPreviousAddress.confirmPage.changeLinkText = Newid cyfeiriad c
 
 authorisedOfficialPreviousAddress.checkYourAnswersLabel = Cyfeiriad cartref blaenorol
 
+# Amend Authorised Official Previous Address Messages
+# ----------------------------------------------------------
+amendAuthorisedOfficialPreviousAddress.title = Diwygio cyfeiriad cartref blaenorol y swyddog awdurdodedig - Swyddogion ac enwebeion
```
Comment (yangwong9664) [inline conf/messages.cy:1214]: should `- Swyddogion ac enwebeion` be there?

**Review by yangwong9664 — CHANGES_REQUESTED** [review]

File: conf/messages.cy (hunk 2 above)
Comment (anburoach) [reply conf/messages.cy:1214]: yes.. section name should be part of the title.. so I added that

File: test/navigation/AuthorisedOfficialsNavigatorSpec.scala [hunk 3]
```diff
@@ -204,6 +205,14 @@ class AuthorisedOfficialsNavigatorSpec extends SpecBase {
                 .success.value) mustBe
               authOfficialRoutes.IsAuthorisedOfficialPreviousAddressController.onPageLoad(NormalMode, index)
           }
+
+          "go to the Amend address page if one or more address lines length >35 characters when clicked Confirm and continue button" in {
```
Comment (yangwong9664) [inline test/navigation/AuthorisedOfficialsNavigatorSpec.scala:209]: can we get a test for special characters too

Comment (anburoach) [reply test/navigation/AuthorisedOfficialsNavigatorSpec.scala:209]: All amend address pages using same html and form provider, so I wrote test for special characters in 
[https://github.com/hmrc/charities-registration-frontend/pull/439/files#diff-29efb6569bd2837394c1f62964d46176d16940d8ded36782d8a8e58ca8ebcac1R140]

**Review by yangwong9664 — APPROVED** [review]

**Pull request merged**